packages/@aws-cdk/aws-lambda/lib/docker.ts

```diff
@@ -0,0 +1,141 @@
+import { spawnSync } from 'child_process';
+
+/**
+ * A Docker volume
+ */
+export interface DockerVolume {
+  /**
+   * The path to the file or directory on the host machine
+   */
+  readonly hostPath: string;
+
+  /**
+   * The path where the file or directory is mounted in the container
+   */
+  readonly containerPath: string;
+}
+
+/**
+ * Docker run options
+ */
+export interface DockerRunOptions {
+  /**
+   * The command to run in the container.
+   */
+  readonly command: string[];
+
+  /**
+   * Docker volumes to mount.
+   *
+   * @default - no volumes are mounted
+   */
+  readonly volumes?: DockerVolume[];
+
+  /**
+   * The environment variables to pass to the container.
+   *
+   * @default - no environment variables.
+   */
+  readonly environment?: { [key: string]: string; };
+
+  /**
+   * Working directory inside the container.
+   *
+   * @default - image default
+   */
+  readonly workingDirectory?: string;
+}
+
+/**
+ * Docker build options
+ */
+export interface DockerBuildOptions {
+  /**
+   * Build args
+   *
+   * @default - no build args
+   */
+  readonly buildArgs?: { [key: string]: string };
+}
+
+/**
+ * A Docker image
+ */
+export class DockerImage {
+  /**
+   * Use an existing Docker image
+   *
+   * @param image the image name
+   */
+  public static fromImage(image: string) {
```

Comment:
I am thinking that maybe we should model this like [`ecs.ContainerImage`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ecs.ContainerImage.html)

1. `fromImage` => `fromRegistry`
2. `fromBuild` => `fromAsset`

Reply:
And the question will be: do we want to also add `fromEcrRepository`, but we can add that later.

Reply:
> we should model this like ecs.ContainerImage

done

> do we want to also add `fromEcrRepository`

for another PR I would say